tests/Integration/Webservice/AddressesTest.php

```diff
@@ -0,0 +1,315 @@
+
+        self::assertEquals(1, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(21, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'No permission for this authentication key',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'POST', 'addresses');
+
+        self::assertEquals(1, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(21, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'No permission for this authentication key',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'PUT', 'addresses');
+
+        self::assertEquals(1, $output->filter('prestashop > errors > error')->count());
+        self::assertEquals(21, $output->filter('prestashop > errors > error')->first()->filter('code')->text());
+        self::assertEquals(
+            'No permission for this authentication key',
+            $output->filter('prestashop > errors > error')->first()->filter('message')->text()
+        );
+    }
+
+    public function testReturnGet(): void
+    {
+        // Set access
+        WebserviceKey::setPermissionForAccount(
+            $this->wsKey->id,
+            [
+                'addresses' => [
+                    'GET' => 'on',
+                ],
+            ]
+        );
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'GET', 'addresses');
+
+        self::assertEquals(self::INITIAL_COUNT, $output->filter('prestashop > addresses > address')->count());
+
+        $id = $output->filter('prestashop > addresses > address')->first()->attr('id');
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'GET', 'addresses/' . $id);
+
+        $address = new Address($id);
+
+        $children = $output->filter('prestashop > address')->children();
+        foreach ($children as $item) {
+            /** @var \DOMElement $item */
+            $property = $item->nodeName;
+            self::assertEquals($address->{$property}, $item->nodeValue);
+        }
+    }
+
+    public function testManipulateData(): void
+    {
+        // Set access
+        WebserviceKey::setPermissionForAccount(
+            $this->wsKey->id,
+            [
+                'addresses' => [
+                    'GET' => 'on',
+                    'DELETE' => 'on',
+                    'PUT' => 'on',
+                    'POST' => 'on',
+                ],
+            ]
+        );
+
+        // Add an object in database
+        $data = '<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
```

Comment:
In order to avoid repeating these custom XML document when you add more tests, maybe a function `generateInputXMLForAddress()` could be used?